### ISSUE TITLE
Issue regarding Katib sample random-example Failed

### ISSUE THREAD

**Issue by subhabratabanerjee** [kubeflow/katib#1550]
/kind bug

**What steps did you take and what happened:**

I am runnings Katib sample random-example- https://github.com/kubeflow/katib/blob/master/examples/v1beta1/random-example.yaml,  with adding sidecar.istio.io/inject: “false"
in yaml I am using training-container (without any change) still in log showing 'choose one of: [training-container metrics-logger-and-collector]’

added pod describe & logs

pod describe - 

$>k-n describe po random-example-subho-2rmwkwx6-hkdwb

Name:               random-example-subho-2rmwkwx6-hkdwb
Namespace:          kubeflow-user-example-com
Priority:           0
PriorityClassName:  <none>
Node:               XX.XX.XXX.XX/XX.XX.XXX.XX
Start Time:         Fri, 04 Jun 2021 04:14:07 -0700
Labels:             controller-uid=a292bd81-6856-41c0-a4f4-19c0e17c4621
                    job-name=random-example-subho-2rmwkwx6
Annotations:        cni.projectcalico.org/podIP: Xy.XX.XXX.Xz/32
                    cni.projectcalico.org/podIPs: Xy.XX.XXX.Xz/32
                    sidecar.istio.io/inject: false
Status:             Failed
IP:                 Xy.XX.XXX.Xz
Controlled By:      Job/random-example-subho-2rmwkwx6
Containers:
  training-container:
    Container ID:  docker://de406eeeddce8200b5fa097227a19c8975bf292b410910ff531ec50f5f797a02
    Image:         docker.io/kubeflowkatib/mxnet-mnist:v1beta1-45c5727
    Image ID:      docker-pullable://kubeflowkatib/mxnet-mnist@sha256:9bbfc47d1fc369e79d0b4e83f26b3060941eb0d0792c758a4ce27b4bd90a6c48
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      -c
    Args:
      python3 /opt/mxnet-mnist/mnist.py --batch-size=64 --lr=0.016855643743925692 --num-layers=3 --optimizer=adam 1>/var/log/katib/metrics.log 2>&1 && echo completed > /var/log/katib/$$$$.pid
    State:          Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Fri, 04 Jun 2021 04:14:09 -0700
      Finished:     Fri, 04 Jun 2021 04:25:04 -0700
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/log/katib from metrics-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-c8vl4 (ro)
  metrics-logger-and-collector:
    Container ID:  docker://5cf990b5caf0c3abc49d20e60331563abd401904ec1bdacc57620111deadcfd2
    Image:         docker.io/kubeflowkatib/file-metrics-collector:v0.11.0
    Image ID:      docker-pullable://kubeflowkatib/file-metrics-collector@sha256:6aff0f5c41489aa04f387a8e127607cdfd7f3168a565840e1fc43cd2a4c1401c
    Port:          <none>
    Host Port:     <none>
    Args:
      -t
      random-example-subho-2rmwkwx6
      -m
      Validation-accuracy;Train-accuracy
      -o-type
      maximize
      -s-db
      katib-db-manager.kubeflow:6789
      -path
      /var/log/katib/metrics.log
    State:          Terminated
      Reason:       Error
      Exit Code:    255
      Started:      Fri, 04 Jun 2021 04:14:10 -0700
      Finished:     Fri, 04 Jun 2021 04:25:05 -0700
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:                500m
      ephemeral-storage:  5Gi
      memory:             100Mi
    Requests:
      cpu:                50m
      ephemeral-storage:  500Mi
      memory:             10Mi
    Environment:          <none>
    Mounts:
      /var/log/katib from metrics-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-c8vl4 (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  default-token-c8vl4:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-c8vl4
    Optional:    false
  metrics-volume:
    Type:        EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:      
    SizeLimit:   <unset>
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason            Age                 From                   Message
  ----     ------            ----                ----                   -------
  Normal   Scheduled         <unknown>           default-scheduler      Successfully assigned kubeflow-user-example-com/random-example-subho-2rmwkwx6-hkdwb to XX.XX.XXX.XX
  Normal   Pulled            27m                 kubelet, XX.XX.XXX.XX  Container image "docker.io/kubeflowkatib/mxnet-mnist:v1beta1-45c5727" already present on machine
  Normal   Created           27m                 kubelet, XX.XX.XXX.XX  Created container training-container
  Normal   Started           27m                 kubelet, XX.XX.XXX.XX  Started container training-container
  Normal   Pulling           27m                 kubelet, XX.XX.XXX.XX  Pulling image "docker.io/kubeflowkatib/file-metrics-collector:v0.11.0"
  Normal   Created           27m                 kubelet, XX.XX.XXX.XX  Created container metrics-logger-and-collector
  Normal   Pulled            27m                 kubelet, XX.XX.XXX.XX  Successfully pulled image "docker.io/kubeflowkatib/file-metrics-collector:v0.11.0"
  Normal   Started           27m                 kubelet, XX.XX.XXX.XX  Started container metrics-logger-and-collector

pod log

$> k-n logs  random-example-subho-2rmwkwx6-hkdwb
Error from server (BadRequest): a container name must be specified for pod random-example-subho-2rmwkwx6-hkdwb, choose one of: [training-container metrics-logger-and-collector]

**What did you expect to happen:**

should run with out any issue

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Kubeflow version (`kfctl version`): 1.3
- Minikube version (`minikube version`):
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

**Comment by andreyvelich** [comment]
If you want to check the logs from `random-example-subho-2rmwkwx6-hkdwb` pod you have to specify the container name since pod has 2 containers.
For example:
`kubectl logs random-example-subho-2rmwkwx6-hkdwb -n kubeflow-user-example-com -c metrics-logger-and-collector`
or 
`kubectl logs random-example-subho-2rmwkwx6-hkdwb -n kubeflow-user-example-com -c training-container`

**Comment by subhabratabanerjee** [comment]
@andreyvelich , thanks for pointing this out , seems like a proxy issue that’s why not able to connect, as Katib is ml f/w agnostics can I run a sklearn also ? (I am a newbie in Katib ) in documentation I could’t find about sklearn that why I am asking

021-06-04T17:53:42Z DEBUG    Starting new HTTP connection (1): data.mxnet.io:80
I0604 17:55:54.029132      69 main.go:136] 2021-06-04T17:55:54Z DEBUG    Starting new HTTP connection (1): data.mxnet.io:80
I0604 17:58:05.101030      69 main.go:136] 2021-06-04T17:58:05Z DEBUG    Starting new HTTP connection (1): data.mxnet.io:80
I0604 18:00:16.173006      69 main.go:136] 2021-06-04T18:00:16Z DEBUG    Starting new HTTP connection (1): data.mxnet.io:80

**Comment by andreyvelich** [comment]
@subhabratabanerjee Yes, you just need to build Docker image for your training container code and you should be able to run it in the Experiment.
